src/casanovoutils/mgfutils.py

```
@@ -342,10 +342,169 @@ def pipeline(
+
+    Returns
+    -------
+    list[PyteomicsSpectrum]
+        Sampled spectra, grouped by peptide in first-seen order.
+    """
+    configure_logging(pathlib.Path(outfile).with_suffix(".log") if outfile else None)
+    logging.info(
+        "Sampling up to k=%d spectra per peptide (random_seed=%d)", k, random_seed
+    )
+
+    rng = random.Random(random_seed)
```

Ruff (0.15.9)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)